osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -307,6 +307,17 @@ private ArchiveReader getReaderFrom(string path)
         /// <returns>The imported beatmap, or an existing instance if it is already present.</returns>
         private BeatmapSetInfo importToStorage(ArchiveReader reader)
         {
+            string mapName;
+            // let's make sure there are actually .osu files to import.
+            try
+            {
+                mapName = reader.Filenames.First(f => f.EndsWith(".osu"));

```

Comment:
How about something like:
```
string mapName = reader.Filenames.FirstOrDefault(f => f.EndsWith(".osu"));
if (string.IsNullOrEmpty(mapName))
    throw new InvalidOperationException(...);
```
To avoid the try catch.